wavelength-plugins/icons.js

```diff
@@ -1,84 +1,87 @@
-/*
-* icons.js
-* Credits: Lord Haji, panpawn
-*/
-'use strict';
+/******************************
+* Icons for Pokemon Showdown  *
+* Credits: Lord Haji, panpawn.*
+*******************************/
+"use strict";
 
-let icons = {};
-const fs = require('fs');
+const FS = require("../lib/fs.js");
 
-function load() {
-	fs.readFile('config/icons.json', 'utf8', function (err, file) {
-		if (err) return;
-		icons = JSON.parse(file);
-	});
+let icons = FS("config/icons.json").readIfExistsSync();
+
+if (icons !== "") {
+	icons = JSON.parse(icons);
+} else {
+	icons = {};
 }
-load();
 
 function updateIcons() {
-	fs.writeFileSync('config/icons.json', JSON.stringify(icons));
+	FS("config/icons.json").writeUpdate(() => (
+		JSON.stringify(icons)
+	));
 
-	let newCss = '/* ICONS START */\n';
+	let newCss = "/* ICONS START */\n";
 
 	for (let name in icons) {
 		newCss += generateCSS(name, icons[name]);
 	}
-	newCss += '/* ICONS END */\n';
+	newCss += "/* ICONS END */\n";
 
-	let file = fs.readFileSync('config/custom.css', 'utf8').split('\n');
-	if (~file.indexOf('/* ICONS START */')) file.splice(file.indexOf('/* ICONS START */'), (file.indexOf('/* ICONS END */') - file.indexOf('/* ICONS START */')) + 1);
-	fs.writeFileSync('config/custom.css', file.join('\n') + newCss);
+	let file = FS("config/custom.css").readIfExistsSync().split("\n");
+	if (~file.indexOf("/* ICONS START */")) file.splice(file.indexOf("/* ICONS START */"), (file.indexOf("/* ICONS END */") - file.indexOf("/* ICONS START */")) + 1);
+	FS("config/custom.css").writeUpdate(() => (
+		file.join("\n") + newCss
+	));
 	WL.reloadCSS();
 }
 
 function generateCSS(name, icon) {
-	let css = '';
-	let rooms = [];
+	let css = "";
 	name = toId(name);
-	Rooms.rooms.forEach((curRoom, id) => {
-		if (curRoom.id === 'global' || curRoom.type !== 'chat' || curRoom.isPersonal) return;
-		if (!isNaN(Number(id.charAt(0)))) return;
-		rooms.push('#' + id + '-userlist-user-' + name);
-	});
-	css = rooms.join(', ');
-	css += '{\nbackground: url("' + icon + '") no-repeat right\n}\n';
+	css = `[id$="-userlist-user-${name}] {\nbackground: url("${icon}") no-repeat right !important;\n}\n`;
```

Comment:
Thank *God* for changing this. Oh my god. The amount of CSS reduced...you're doing God's work, kiddo.

Reply:
Thanks LOL